src/frontend/src/handler/create_table.rs

```diff
@@ -1320,12 +1349,334 @@ pub async fn handle_create_table(
+    );
+    with.insert("catalog.name".to_string(), iceberg_catalog_name.clone());
+    with.insert("database.name".to_string(), iceberg_database_name.clone());
+    with.insert("table.name".to_string(), iceberg_table_name.to_string());
+    with.insert("commit_checkpoint_interval".to_string(), "1".to_string());
+    with.insert("create_table_if_not_exists".to_string(), "true".to_string());
+    with.insert("enable_config_load".to_string(), "true".to_string());
+    sink_handler_args.with_options = WithOptions::new_with_options(with);
+
+    let mut source_name = table_name.clone();
+    *source_name.0.last_mut().unwrap() = Ident::from(
+        (ICEBERG_SOURCE_PREFIX.to_string() + &source_name.0.last().unwrap().real_value()).as_str(),
+    );
+    let create_source_stmt = CreateSourceStatement {
+        temporary: false,
+        if_not_exists: false,
+        columns: vec![],
+        source_name,
+        wildcard_idx: None,
+        constraints: vec![],
+        with_properties: WithProperties(vec![]),
+        format_encode: CompatibleFormatEncode::V2(FormatEncodeOptions::none()),
+        source_watermarks: vec![],
+        include_column_options: vec![],
+    };
+
+    let mut source_handler_args = handler_args.clone();
+    let mut with = BTreeMap::new();
+    with.insert("connector".to_string(), "iceberg".to_string());
+    with.insert("catalog.type".to_string(), "jdbc".to_string());
+    with.insert("warehouse.path".to_string(), warehouse_path.clone());
+    if let Some(s3_endpoint) = s3_endpoint {
+        with.insert("s3.endpoint".to_string(), s3_endpoint.clone());
+    }
+    if let Some(s3_ak) = s3_ak.clone() {
+        with.insert("s3.access.key".to_string(), s3_ak.clone());
+    }
+    if let Some(s3_sk) = s3_sk.clone() {
+        with.insert("s3.secret.key".to_string(), s3_sk.clone());
+    }
+    with.insert("s3.region".to_string(), s3_region.clone());
+    with.insert("catalog.uri".to_string(), catalog_uri.clone());
+    with.insert("catalog.jdbc.user".to_string(), meta_store_user.clone());
+    with.insert(
+        "catalog.jdbc.password".to_string(),
+        meta_store_password.clone(),
+    );
+    with.insert("catalog.name".to_string(), iceberg_catalog_name.clone());
+    with.insert("database.name".to_string(), iceberg_database_name.clone());
+    with.insert("table.name".to_string(), iceberg_table_name.to_string());
+    with.insert("enable_config_load".to_string(), "true".to_string());
+    source_handler_args.with_options = WithOptions::new_with_options(with);
+
+    // before we create the table, ensure the JVM is initialized as we use jdbc catalog right now.
+    // If JVM isn't initialized successfully, current not atomic ddl will result in a partially created iceberg engine table.
+    let _ = JVM.get_or_init()?;
+
     let catalog_writer = session.catalog_writer()?;
+    // TODO(iceberg): make iceberg engine table creation ddl atomic
     catalog_writer
         .create_table(source, table, graph, job_type)
```

Comment:
Here the `source` is passed as well, can it work? 🤔

What I am thinking is that, for a common table with connector, the corresponding `source` is supposed to generate changes that will be applied to the `table`. But here the Iceberg source is just for batch read, which I think is actually irrelevant/unconnected to the iceberg table internally.

Reply:
The source here is something like `kafka` and `postgres cdc` instead of the iceberg source.  For example
```
create table t (a int) with (connector = 'kafka' ...) engine = iceberg.
```